src/common.js

```diff
@@ -171,6 +171,11 @@ function parseOptions(opt, map, errorOptions) {
     throw new Error('parseOptions() internal error: errorOptions must be object');
   }
 
+  if (opt === '--') {
+    // This means there are no options.
+    return {};
+  }
```

Comment:
Given my realization above, what we need to do here is just ignore the next argument if `'--'` is present. This could be a bit tricky to implement though, looking at the way `parseOptions` works.

Reply:
No, there's at-most 1 option string in shelljs. That means that `cp('-rf', 'foo', 'dir')` is different than `cp('-r', '-f', 'foo', 'dir')`. The former copies one dir recursively and forcibly, the latter recursively copies 2 dirs named `-f` and `foo` into `dir`.